Gui View and Event Handling

sarahabimay opened the pull request:
This is my 'get it to work' code. Includes:
- Simplified GUI without GameType and Game Size
- Event Handling for Board buttons and replay button.
- Disable buttons when Win or Draw announced
- Replay button only displayed after win or draw announced

I think for the 'make it right' code, the GUIDisplay class feels like I could extract out the code related to the 'DisplayBoard' into it's own class and make it a dependency of GUIDisplay.  
However, by doing this I would have 3 classes with JavaFx code: 
- DisplayBoard would need to create javafx Buttons for the board before wrapping them in my JavaFxButton.
- GUIDisplay has the most dependencies on JavaFx elements.
- Main class extends javafx Application class.

What do you think?

@Maikon
@jsuchy

sarahabimay committed:
WIP : GUI Event Handling

sarahabimay committed:
Merge branch 'master' into gui_view

sarahabimay committed:
Methods for displaying and modifying GUI board

sarahabimay committed:
Event handler tests.  Pass id of button clicked to handler.

sarahabimay committed:
TDD of Controller class.

sarahabimay committed:
Display results. Disable board after win. Display replay button.

sarahabimay committed:
Add Event handling for PlayAgain? button.

sarahabimay committed:
Create test/javafxgui package. Move methods around. Remove unnecessary spy.

sarahabimay committed:
Extract some methods to create JavaFx elements

sarahabimay committed:
Move method from TTTController into GUIDisplay

sarahabimay committed:
Test added to test displayPlayAgain method in TTTController

gemcfadyen commented on GUI/test/javafxgui/ControllerTest.java:
```diff
@@ -0,0 +1,167 @@
+
+    @Test
+    public void tellGUIDisplayToDisableBoard() {
+        Mark X = Mark.X;
```
Why do you need to declare X and O here? You can just use Mark.X and Mark.O directly when initialising the board, and do a static import, so it would look just like what you have for currentBoard.

sarahabimay committed:
Remove GameFake as serving no purpose.

sarahabimay committed:
Extract a BoardDisplay class

sarahabimay committed:
Extract EventRegister class to handle Button to EventHandler registration.

The first two tests.
k

sarahabimay committed:
Test method which registers all ClickableElements with appropriate Handler.

Just extracting Board buttons initially.

sarahabimay committed:
TTTController is now responsible for passing Scene or GameBoard to EventRegister.

sarahabimay committed:
Add test for method which wasn't covered.

sarahabimay committed:
Register event handling for PlayAgain button from TTTController

sarahabimay committed:
Fixed issue with event handling for PlayAgain button.

When play again is chosen, the Game instance was being replaced which therefore created
a new Board instance too.  However BoardDisplay held a dependency on the Board instance
and this wasn't being replaced when the board was re-instantiated.
This code is mainly to fix that issue.
